### PR TITLE
add offline/online status

### PR DESCRIPTION
**Pull request by hutnikau**
depends on https://github.com/oat-sa/extension-tao-act/pull/225

### CODE REVIEW

File: helpers/DeliveryHelper.php [hunk 1]
```diff
@@ -33,6 +33,7 @@
 use oat\taoProctoring\model\monitorCache\implementation\DeliveryMonitoringService;
 use oat\taoQtiTest\models\event\QtiTestChangeEvent;
 use qtism\runtime\tests\AssessmentTestSessionState;
+use oat\taoAct\model\runner\HeartbeatService;
```
Comment (jsconan) [inline helpers/DeliveryHelper.php]: Please do not directly use the heartbeat here, as it is a customer specific. You should rather use an abstraction to handle the last activity. Thus an implementation of this abstraction can be done to handle the heartbeat, and this implementation should be registered from a customer extension.

File: helpers/DeliveryHelper.php [hunk 2]
```diff
@@ -455,68 +456,11 @@ private static function adjustDeliveryExecutions($deliveryExecutions, $options)
...
-                    )
-                ),
-            );
-            /********/
-
             $executions = array();
             
-            
             $deliveryMonitoringService = ServiceManager::getServiceManager()->get(DeliveryMonitoringService::CONFIG_ID);
+            $heartbeatService = ServiceManager::getServiceManager()->get(HeartbeatService::SERVICE_ID);
```
Comment (jsconan) [inline helpers/DeliveryHelper.php]: You should use an abstraction, not the heartbeat itself.

File: views/js/controller/Delivery/monitoring.js [hunk 3]
```diff
@@ -426,13 +426,20 @@ define([
                 id: 'status',
                 label: __('Status'),
                 transform: function(value, row) {
-                    if(row && row.state && row.state.status){
-                        var status = _status.getStatusByCode(row.state.status);
-                        if(status){
-                            return status.label;
+                    var result = '',
+                        status,
+                        online = row.online ? 'online' : 'offline';
```
Comment (jsconan) [inline views/js/controller/Delivery/monitoring.js]: This label should be translated

**Comment by hutnikau** [comment]
@jsconan the branch has been updated according to your comments. Please review it again.

**Comment by hutnikau** [comment]
Please note that it depends on https://github.com/oat-sa/extension-tao-act/pull/227

File: manifest.php [hunk 4]
```diff
@@ -24,7 +24,7 @@
 	'label' => 'Proctoring',
 	'description' => 'Proctoring for deliveries',
     'license' => 'GPL-2.0',
-    'version' => '1.12.2',
+    'version' => '1.12.3',
```
Comment (jsconan) [inline manifest.php]: Not sure it is a fix! It seems rather to be a feature...

Comment (hutnikau) [inline manifest.php]: Fixed!

**Pull request merged**